crates/transaction-pool/src/validate/eth.rs

```diff
@@ -22,27 +22,34 @@ use tokio::sync::{oneshot, Mutex};
 ///
 /// This validator is non-blocking, all validation work is done in a separate task.
 #[derive(Debug, Clone)]
+pub struct TransactionValidationTaskExecutor<V> {
+    /// The validator that will validate transactions on a separate task.
+    validator: V,
+    /// The sender half to validation tasks that perform the actual validation.
+    to_validation_task: Arc<Mutex<ValidationJobSender>>,
+}
+
+/// Validator for Ethereum transactions.
+#[derive(Debug)]
 pub struct EthTransactionValidator<Client, T> {
     /// The type that performs the actual validation.
     inner: Arc<EthTransactionValidatorInner<Client, T>>,
-    /// The sender half to validation tasks that perform the actual validation.
-    to_validation_task: Arc<Mutex<ValidationJobSender>>,
 }
 
-// === impl EthTransactionValidator ===
+// === impl TransactionValidationTaskExecutor ===
 
-impl EthTransactionValidator<(), ()> {
+impl TransactionValidationTaskExecutor<()> {
     /// Convenience method to create a [EthTransactionValidatorBuilder]
     pub fn builder(chain_spec: Arc<ChainSpec>) -> EthTransactionValidatorBuilder {
         EthTransactionValidatorBuilder::new(chain_spec)
     }
 }
 
-impl<Client, Tx> EthTransactionValidator<Client, Tx> {
+impl<Client, Tx> TransactionValidationTaskExecutor<EthTransactionValidator<Client, Tx>> {
```

Comment:
this should be generic over V: TransactionValidator + Clone instead

Reply:
@mattsse Done